File: find.dart
void main() {
  var arr = [4, 5, 4, 6, 3, 4, 5, 2, 23, 1, 4, 4, 4];
  find(arr);

  arr = [4, 2];
  find(arr);

  arr = [2];
  find(arr);
}

void find(List arr) {
  // print original
  print('[*] original array: $arr');

  // validate
  if (arr.length < 2) {
    print('Invalid array');
    return;
  }

  // unique
  List unique = arr.toSet().toList();

  // sort
  unique.sort();

  // print sorted
  print('[*] unique and sorted: $unique');

  // print results
  print('[*] secondSmallest: ${unique[1]}');
  print('[*] secondLargest: ${unique[unique.length - 2]}');
}

File: Find.java
import java.util.*;

class Find {

  public static void main(String[] args) {
    Integer[] arr = { 1, 2, 4, 10, 20, 20, 2, 4 };
    find(arr);
  }

  public static void find(Integer[] arr) {
    // Validate
    if (arr.length < 2) {
      System.out.println("array is invalid");
    }

    // Unique & Sorted
    Set<Integer> unique = new TreeSet<>();
    Collections.addAll(unique, arr);
    System.out.println(unique);

    Integer[] sorted = {};
    sorted = unique.toArray(sorted);

    Integer secondSmallest = sorted[1];
    Integer secondLargest = sorted[sorted.length - 2];

    System.out.println("secondSmallest: " + secondSmallest + ", secondLargest:" + secondLargest);
  }
}